Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -8,52 +9,141 @@
+            var manifestResponse = await GetManifest(matchingManifest.digest);
+            return await TryReadSingleImage(await manifestResponse.Content.ReadFromJsonAsync<ManifestV2>());
+        }
 
-        if (manifest is null) throw new NotImplementedException("Got a manifest but it was null");
+        async Task<HttpResponseMessage> GetManifest(string reference)
+        {
+            var client = GetClient();
+            var response = await client.GetAsync(new Uri(BaseUri, $"/v2/{name}/manifests/{reference}"));
+            response.EnsureSuccessStatusCode();
+            return response;
+        }
 
-        if ((string?)manifest["mediaType"] != DockerManifestV2)
+        async Task<HttpResponseMessage> GetBlob(string digest)
         {
-            throw new NotImplementedException($"Do not understand the mediaType {manifest["mediaType"]}");
+            var client = GetClient();
+            var response = await client.GetAsync(new Uri(BaseUri, $"/v2/{name}/blobs/{digest}"));
+            response.EnsureSuccessStatusCode();
+            return response;
         }
+    }
 
-        JsonNode? config = manifest["config"];
-        Debug.Assert(config is not null);
-        Debug.Assert(((string?)config["mediaType"]) == DockerContainerV1);
+    private string? CheckIfRidExistsInGraph(RuntimeGraph graphForManifestList, string userRid) => graphForManifestList.Runtimes.FirstOrDefault(kvp => graphForManifestList.AreCompatible(kvp.Key, userRid)).Key;
 
-        string? configSha = (string?)config["digest"];
-        Debug.Assert(configSha is not null);
+    private (IReadOnlyDictionary<string, PlatformSpecificManifest>, RuntimeGraph) ConstructRuntimeGraphForManifestList(ManifestListV2 manifestList, RuntimeGraph dotnetRuntimeGraph)
+    {
+        var ridDict = new Dictionary<string, PlatformSpecificManifest>();
+        var runtimeDescriptionSet = new HashSet<RuntimeDescription>();
+        foreach (var manifest in manifestList.manifests) {
+            if (CreateRidForPlatform(manifest.platform) is { } rid)
+            {
+                if (ridDict.TryAdd(rid, manifest)) {
+                    AddRidAndDescendantsToSet(runtimeDescriptionSet, rid, dotnetRuntimeGraph);
+                }
+            }
+        }
+        
+        var graph = new RuntimeGraph(runtimeDescriptionSet);
+        return (ridDict, graph);
+    }
 
-        response = await client.GetAsync(new Uri(BaseUri, $"/v2/{name}/blobs/{configSha}"));
+    private string? CreateRidForPlatform(PlatformInformation platform)
+    {
+        var osPart = platform.os switch
+        {
+            "linux" => "linux",
+            "windows" => "win",
+            _ => null
+        };
+        // TODO: this part needs a lot of work, the RID graph isn't super precise here and version numbers (especially on windows) are _whack_
+        // TODO: we _may_ need OS-specific version parsing. Need to do more research on what the field looks like across more manifest lists.
```

Comment:
I'd like to call this part out - Rich says we're leaning away from OS-specific RIDs in general, so we may just be able to drop versions?

Reply:
Worth noting that the version section is what makes us need to parse the actual runtimegraph - because of this the manifest supports win10-x64, but the SDK portable RID for windows is win-x64, so we need the compatibility relationship to work.

Reply:
This seems pretty reasonable to me, at least good enough to get feedback on.